### ISSUE TITLE
Simple example for parse, modify and save.

### ISSUE THREAD

**Issue by pjdurai** [inab/python-groovy-parser#3]
Greetings.

I am new to this. I am looking for an example of loading a groovy file, parse it, modify it a bit (like removing a specific block of code) and save it back.

Is that possible?

Thanks!

**Comment by jmfernandez** [comment]
Hi @pjdurai , it is possible loading a groovy file and parsing it. For instance, https://github.com/inab/python-groovy-parser/blob/main/translated-groovy3-parser.py or https://github.com/inab/python-groovy-parser/blob/main/cached-translated-groovy3-parser.py have a very rough demonstration of it, where some features in the obtained tokenized Lark parse tree from the input Groovy file are detected.

Although there is no code developed to write back the obtained parsed Lark tree (or a manipulated one) to its groovy representation, it should not be too much difficult, as it would be to visit all the [tree leaf nodes](https://github.com/inab/python-groovy-parser/blob/19cad563f2b3dd3bede85d5883d9810920fd6e81/groovy_parser/parser.py#L50) in preorder and write the payload associated to each leaf node.

The only tricky part is that method `analyze_nf_source` does not allow to customize which separators are filtered out [when it calls digest_lark_tree](https://github.com/inab/python-groovy-parser/blob/19cad563f2b3dd3bede85d5883d9810920fd6e81/translated-groovy3-parser.py#L374). The method does not allow either to pass those [customizations to default method from LarkFilteringTreeEncoder](https://github.com/inab/python-groovy-parser/blob/19cad563f2b3dd3bede85d5883d9810920fd6e81/groovy_parser/parser.py#L179). 

So, I am adding the needed parameters with default values, in order to allow you to tell that no separator should be filtered out in your use case.

**Comment by jmfernandez** [comment]
I have just released version 0.1.2 , where I have added the example script [parser-groovy-writer.py](https://github.com/inab/python-groovy-parser/blob/a0839017e3365096ceafc6a4f5685f35a07127b2/parser-groovy-writer.py) which shows a rewritten groovy code from the parse tree. But, as some elements, like comments and some whitespaces in specific places, are never passed from the tokenizer to the lexer, or from the lexer to the parser, they are lost in the generated output. For instance:

```groovy
#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    nf-core/wombat
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/nf-core/wombat
    Website: https://nf-co.re/wombat
    Slack  : https://nfcore.slack.com/channels/wombat
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2


/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    VALIDATE & PRINT PARAMETER SUMMARY
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

WorkflowMain.initialise(workflow, params, log)

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOW FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { WOMBAT } from './workflows/wombat'

//
// WORKFLOW: Run main nf-core/wombat analysis pipeline
//
workflow NFCORE_WOMBAT {
    WOMBAT ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN ALL WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Execute a single named workflow for the pipeline
// See: https://github.com/nf-core/rnaseq/issues/619
//
workflow {
    NFCORE_WOMBAT ()
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
```

is written as

```groovy



nextflow.enable.dsl=2




WorkflowMain.initialise(workflow, params, log)



include {WOMBAT} from './workflows/wombat'




workflow NFCORE_WOMBAT {
WOMBAT()
}







workflow {
NFCORE_WOMBAT()
}


```

So, in case you have to keep them, feel free to fork the project in order to improve the code.

Hope this helps!
